In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
df = pd.read_csv(r"C:\Users\ACER\Downloads\EDAproject\data\raw\Amazon.csv")

df.head()

,OrderID,OrderDate,CustomerID,CustomerName,ProductID,ProductName,Category,Brand,Quantity,UnitPrice,Discount,Tax,ShippingCost,TotalAmount,PaymentMethod,OrderStatus,City,State,Country,SellerID
0,ORD0000001,31-01-2023,CUST001504,Vihaan Sharma,P00014,Drone Mini,Books,BrightLux,3,106.59,0.00,0.00,0.09,319.86,Debit Card,Delivered,Washington,DC,India,SELL01967
1,ORD0000002,30-12-2023,CUST000178,Pooja Kumar,P00040,Microphone,Home & Kitchen,UrbanStyle,1,251.37,0.05,19.10,1.74,259.64,Amazon Pay,Delivered,Fort Worth,TX,United States,SELL01298
2,ORD0000003,10-05-2022,CUST047516,Sneha Singh,P00044,Power Bank 20000mAh,Clothing,UrbanStyle,3,35.03,0.10,7.57,5.91,108.06,Debit Card,Delivered,Austin,TX,United States,SELL00908
3,ORD0000004,18-07-2023,CUST030059,Vihaan Reddy,P00041,Webcam Full HD,Home & Kitchen,Zenith,5,33.58,0.15,11.42,5.53,159.66,Cash on Delivery,Delivered,Charlotte,NC,India,SELL01164
4,ORD0000005,04-02-2023,CUST048677,Aditya Kapoor,P00029,T-Shirt,Clothing,KiddoFun,2,515.64,0.25,38.67,9.23,821.36,Credit Card,Cancelled,San Antonio,TX,Canada,SELL01411


In [3]:
df.shape

(100000, 20)

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 20 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   OrderID        100000 non-null  str    
 1   OrderDate      100000 non-null  str    
 2   CustomerID     100000 non-null  str    
 3   CustomerName   100000 non-null  str    
 4   ProductID      100000 non-null  str    
 5   ProductName    100000 non-null  str    
 6   Category       100000 non-null  str    
 7   Brand          100000 non-null  str    
 8   Quantity       100000 non-null  int64  
 9   UnitPrice      100000 non-null  float64
 10  Discount       100000 non-null  float64
 11  Tax            100000 non-null  float64
 12  ShippingCost   100000 non-null  float64
 13  TotalAmount    100000 non-null  float64
 14  PaymentMethod  100000 non-null  str    
 15  OrderStatus    100000 non-null  str    
 16  City           100000 non-null  str    
 17  State          100000 non-null  str    
 

In [5]:
df.isnull().sum()

OrderID          0
OrderDate        0
CustomerID       0
CustomerName     0
ProductID        0
ProductName      0
Category         0
Brand            0
Quantity         0
UnitPrice        0
Discount         0
Tax              0
ShippingCost     0
TotalAmount      0
PaymentMethod    0
OrderStatus      0
City             0
State            0
Country          0
SellerID         0
dtype: int64

### Missing Value Treatment

No missing values were found in the dataset.
Therefore, no imputation techniques were required.

In [7]:
df.duplicated().sum()

np.int64(0)

### Duplicate Handling

No duplicate records were found in the dataset.
Hence, no duplicate removal was necessary.

In [10]:
df["OrderDate"] = pd.to_datetime(df["OrderDate"], dayfirst=True)

### Data Type Conversion

The OrderDate column was converted from string to datetime format to enable time-based analysis.

In [11]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 20 columns):
 #   Column         Non-Null Count   Dtype         
---  ------         --------------   -----         
 0   OrderID        100000 non-null  str           
 1   OrderDate      100000 non-null  datetime64[us]
 2   CustomerID     100000 non-null  str           
 3   CustomerName   100000 non-null  str           
 4   ProductID      100000 non-null  str           
 5   ProductName    100000 non-null  str           
 6   Category       100000 non-null  str           
 7   Brand          100000 non-null  str           
 8   Quantity       100000 non-null  int64         
 9   UnitPrice      100000 non-null  float64       
 10  Discount       100000 non-null  float64       
 11  Tax            100000 non-null  float64       
 12  ShippingCost   100000 non-null  float64       
 13  TotalAmount    100000 non-null  float64       
 14  PaymentMethod  100000 non-null  str           
 15  OrderStatus 

In [12]:
df["Category"] = df["Category"].str.strip()
df["Brand"] = df["Brand"].str.strip()
df["PaymentMethod"] = df["PaymentMethod"].str.strip()
df["OrderStatus"] = df["OrderStatus"].str.strip()
df["City"] = df["City"].str.strip()
df["State"] = df["State"].str.strip()
df["Country"] = df["Country"].str.strip()

### Categorical Data Cleaning

Leading and trailing spaces were removed from categorical columns such as Category, Brand, PaymentMethod, OrderStatus, City, State, and Country to ensure consistency.

In [13]:
Q1 = df["TotalAmount"].quantile(0.25)
Q3 = df["TotalAmount"].quantile(0.75)

IQR = Q3 - Q1

lower = Q1 - 1.5 * IQR
upper = Q3 + 1.5 * IQR

outliers = df[(df["TotalAmount"] < lower) | (df["TotalAmount"] > upper)]

outliers.shape

(1360, 20)

### Outlier Detection and Treatment

Outliers in the TotalAmount column were identified using the IQR method.

Records outside the acceptable range were removed to improve data quality and ensure accurate analysis.

In [14]:
df = df[(df["TotalAmount"] >= lower) & (df["TotalAmount"] <= upper)]

In [15]:
df.shape

(98640, 20)

### Final Dataset After Cleaning

After removing outliers, the dataset size was reduced from 100,000 rows to 98,640 rows, ensuring better data quality.

In [16]:
df.to_csv(r"C:\Users\ACER\Downloads\EDAproject\data\interim\cleaned_day2.csv", index=False)